torch/onnx/_internal/fx/passes/shape_inference.py

```diff
@@ -30,6 +29,10 @@ def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
         # 2. run FakeTensorProp
         # If (1) and (2) are done with difference FakeTensorMode's, undefined behavior may
         # happen.
+        import torch._dynamo  # NOTE: Do not import at top level.
+
+        # Even torch/__init__.py does it internally, only
+        # Causes circular when torch._dynamo.* surfaces public facing API during `import torch`
```

Comment:
True but not necessary for files not intended to be imported during `import torch` right? 

Why does this file need to be imported so early?

Reply:
no sure.

before this PR, it was being globally imported in the beginning of the module. This change delays to only when it is going to be used